src/components/datagrid/data_grid.tsx

```diff
@@ -299,6 +299,9 @@ function useDefaultColumnWidth(
 
     const widthToFill = gridWidth - claimedWidth;
     const unsizedColumnCount = columns.length - columnsWithWidths.length;
+    if (unsizedColumnCount === 0) {
+      return 0;
```

Comment:
```suggestion
      return 100;
```

In keeping with the idea that the minimum width is `100`. Though, in this case, the value logically does not matter, the functional idea is that `defaultColumnWidth`, when set, is always a >0 value. Given the logic on line https://github.com/elastic/eui/blob/5318c2f295b35fdbe5ea30fc41249cfbaa2cdfbd/src/components/datagrid/data_grid.tsx#L942, returning `0` prevents the grid from ever rendering.